packages/lambda-xyz/src/tile.set.cache.ts

```diff
@@ -9,20 +16,115 @@ export function getTileSet(name: string, projection: Epsg): TileSet | undefined
     return TileSets.get(tileSetId);
 }
 
-export async function loadTileSet(req: LambdaContext, name: string, projection: Epsg): Promise<TileSet | null> {
-    req.set('tileSet', name);
-    req.set('projection', projection);
+function titleizeName(name: string): string {
+    return name.replace(/_0-/g, ' 0.').replace(/_/g, ' ');
+}
+
+function individualTileSet(parent: TileSet, image: TileSetRuleImagery, setId?: string): TileSet {
+    const { id } = image.imagery;
+    if (setId == null) {
+        setId = TileMetadataTable.unprefix(RecordPrefix.Imagery, id);
+    }
+    const copy = new TileSet(setId, parent.projection);
+    // use parent data as prototype for child;
+    copy.tileSet = Object.create(parent.tileSet ?? null);
+
+    copy.titleOverride = `${parent.title} ${titleizeName(image.imagery.name)}`;
+    copy.extentOverride = Bounds.fromJson(image.imagery.bounds);
+
+    const rule = {
+        id,
+        minZoom: 0,
+        maxZoom: image.rule.maxZoom,
+        priority: 0,
+    };
+    copy.tileSet.imagery = { [id]: rule };
+
+    copy.imagery = [
+        {
+            rule,
+            imagery: image.imagery,
+        },
+    ];
+
+    return copy;
+}
+
+export async function loadTileSet(name: string, projection: Epsg): Promise<TileSet | null> {
+    const subsetIndex = name.indexOf(':');
+    const subsetName = subsetIndex == -1 ? '' : name.slice(subsetIndex + 1);
+    if (subsetName !== '') {
+        name = name.slice(0, subsetIndex);
+    }
     let tileSet = getTileSet(name, projection);
     if (tileSet == null) {
         tileSet = new TileSet(name, projection);
         TileSets.set(tileSet.id, tileSet);
     }
-    req.timer.start('tileset:load');
     const loaded = await tileSet.load();
-    req.timer.end('tileset:load');
     if (!loaded) {
         TileSets.delete(tileSet.id);
         return null;
     }
+    if (subsetName !== '') {
+        const image = tileSet.imagery.find((i) => i.imagery.name === subsetName);
+        if (image == null) return null;
+        const subTileSet = individualTileSet(tileSet, image);
+        subTileSet.tileSet.background = undefined;
```

Comment:
does this need to be done here or can it be done inside the `individualTileSet` ?